boards/sparkfun/micromod/micromod_nrf52840.dts

```diff
@@ -0,0 +1,197 @@
+#include <zephyr/dt-bindings/input/input-event-codes.h>
+
+/ {
+	model = "Sparkfun Micrmod nRF52840 board";
+	compatible = "sparkfun,micromod-nrf52840";
+
+	chosen {
+		zephyr,sram = &sram0;
+		zephyr,flash = &flash0;
+		zephyr,code-partition = &slot0_partition;
+		zephyr,ieee802154 = &ieee802154;
+	};
+
+	/* These aliases are provided for compatibility with samples */
+	aliases {
+		led0 = &led0;
+		watchdog0 = &wdt0;
+		spi-flash0 = &w25q128jv;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+		led0: led_0 {
+			gpios = <&gpio0 13 0>;
+			label = "Blue LED";
+		};
+	};
+
+	micromod_header: connector {
+		compatible = "sparkfun,micromod-gpio";
+		#gpio-cells = <2>;
+		gpio-map-mask = <0xffffffff 0xffffffc0>;
+		gpio-map-pass-thru = <0 0x3f>;
+		gpio-map = <0 0 &gpio0 4 0>,	/* A0 */
+			   <1 0 &gpio0 5 0>,	/* A1 */
+			   <2 0 &gpio0 27 0>,	/* D0 */
+			   <3 0 &gpio1 8 0>,	/* D1/CAM_TRIG */
+			   <4 0 &gpio0 15 0>,	/* I2C_INT# */
+			   <5 0 &gpio0 29 0>,	/* G0/BUS0 */
+			   <6 0 &gpio0 3 0>,	/* G1/BUS1 */
+			   <7 0 &gpio1 13 0>,	/* G2/BUS2 */
+			   <8 0 &gpio1 12 0>,	/* G3/BUS3 */
+			   <9 0 &gpio1 11 0>,	/* G4/BUS4 */
+			   <10 0 &gpio0 17 0>,	/* G5/BUS5 */
+			   <11 0 &gpio1 6 0>,	/* G6/BUS6 */
+			   <12 0 &gpio1 4 0>,	/* G7/BUS7 */
+			   <13 0 &gpio1 14 0>,	/* G8 */
+			   <14 0 &gpio0 9 0>,	/* G9/ADC_D-/CAM_HSYNC */
+			   <15 0 &gpio0 10 0>,	/* G10/ADC_D+/CAM_VSYNC */
+			   //<16 NO CONNECTED	/* G11/SWO */
+			   <17 0 &gpio0 20 0>;	/* SPI_CS */
+	};
+};
+
+&gpiote {
+	status = "okay";
+};
+
+&gpio0 {
+	status = "okay";
+};
+
+&gpio1 {
+	status = "okay";
+};
+
+&uicr {
+	nfct-pins-as-gpios;
+	gpio-as-nreset;
+};
+
+&uart0 {
+	compatible = "nordic,nrf-uart";
+	current-speed = <115200>;
+	status = "okay";
+	pinctrl-0 = <&uart0_default>;
+	pinctrl-1 = <&uart0_sleep>;
+	pinctrl-names = "default", "sleep";
+};
+
+&uart1 {
+	compatible = "nordic,nrf-uart";
+	current-speed = <115200>;
+	status = "okay";
+	pinctrl-0 = <&uart1_default>;
+	pinctrl-1 = <&uart1_sleep>;
+	pinctrl-names = "default", "sleep";
+};
+
+&i2c0 {
+	compatible = "nordic,nrf-twi";
+	status = "okay";
+	pinctrl-0 = <&i2c0_default>;
+	pinctrl-1 = <&i2c0_sleep>;
+	pinctrl-names = "default", "sleep";
+};
+
+&i2c1 {
+	compatible = "nordic,nrf-twi";
+	status = "okay";
+	pinctrl-0 = <&i2c1_default>;
+	pinctrl-1 = <&i2c1_sleep>;
+	pinctrl-names = "default", "sleep";
+};
+
+&spi2 {
+	compatible = "nordic,nrf-spi";
+	status = "okay";
+	cs-gpios = <&gpio0 20 GPIO_ACTIVE_LOW>;
+	pinctrl-0 = <&spi0_default>;
+	pinctrl-1 = <&spi0_sleep>;
+	pinctrl-names = "default", "sleep";
+};
+
+&qspi {
+	status = "okay";
+	pinctrl-0 = <&qspi_default>;
+	pinctrl-1 = <&qspi_sleep>;
+	pinctrl-names = "default", "sleep";
+	w25q128jv: w25q128jv@0 {
+		status="okay";
+		compatible = "nordic,qspi-nor";
+		reg = <0>;
+		// Currently, it supports only pp mode
+		writeoc = "pp";
+		// Currently, it supports only fastread and read2o
+		readoc = "fastread";
+		sck-frequency = <DT_FREQ_M(8)>;
```

Comment:
you can run this much faster, 8MHz is the max for the mx25 on the nrf52840dk because that has an ultra-low power mode which limits the speed to 8MHz but the `w25q128jv` datasheet shows it supports much faster speeds (optional change)